syntax = "proto3";

package mypackage;

import "google/protobuf/timestamp.proto";

message NestedMessage {
  string str_val = 1;
  google.protobuf.Timestamp ts_val = 2;
}

enum MyEnum {
  MY_ENUM_UNSPECIFIED = 0;
  FOO = 1;
  BAR = 2;
  BAZ = 3;
}

message Message {
  double double_val = 1;
  float float_val = 2;
  int32 int32_val = 3;
  int64 int64_val = 4;
  uint32 uint32_val = 5;
  uint64 uint64_val = 6;
  sint32 sint32_val = 7;
  sint64 sint64_val = 8;
  fixed32 fixed32_val = 9;
  fixed64 fixed64_val = 10;
  sfixed32 sfixed32_val = 11;
  sfixed64 sfixed64_val = 12;
  bool bool_val = 13;
  string string_val = 14;
  bytes bytes_val = 15;
  MyEnum enum_val = 16;
  google.protobuf.Timestamp timestamp_val = 17;

  repeated int32 repeated_int32_val = 18;
  repeated NestedMessage repeated_msg_val = 19;

  map<string, NestedMessage> str_to_msg_map = 20;
}
